cpp/include/cudf/io/json.hpp

```diff
@@ -429,6 +491,51 @@ class json_reader_options {
    * @param val An enum value to indicate the JSON reader's behavior on invalid JSON lines.
    */
   void set_recovery_mode(json_recovery_mode_t val) { _recovery_mode = val; }
+
+  /**
+   * @brief Set whether strict validation is enabled or not.
+   *
+   * @param val Boolean value to indicate whether strict validation is enabled.
+   */
+  void set_strict_validation(bool val) { _strict_validation = val; }
+
+  /**
+   * @brief Set whether leading zeros are allowed in numeric values. Strict validation
+   * must be enabled for this to work.
+   *
+   * @note: strict_validation must be enabled for this to work.
+   *
+   * @param val Boolean value to indicate whether leading zeros are allowed in numeric values
+   */
+  void allow_numeric_leading_zeros(bool val) { _allow_numeric_leading_zeros = val; }
```

Comment:
For options like `allow_numeric_leading_zeros` that require `strict_validation` to be set, should we consider `CUDF_EXPECTS( !val || _strict_validation );`?

Are we not doing this because we don't want to require `strict_validation` to be set before `allow_numeric_leading_zeros`?

Reply:
We didn't do it because I didn't think about it. There are other options on how to set things so the APIs are cleaner. https://github.com/rapidsai/cudf/pull/15968#discussion_r1747139998

@karthikeyann do you want me to put together a PR to update how the configs are set, or do you want to try and do that yourself?

Reply:
I will add `CUDF_EXPECTS(_strict_validation)` to these 3 options set functions.